core/block.js

```diff
@@ -278,6 +280,12 @@ Blockly.Block.prototype.dispose = function(healStack) {
     }
   } finally {
     Blockly.Events.enable();
+    if (variableField) {
```

Comment:
It is after the enable for two reasons:
1. I want the VAR_DELETE event to be fired if it happens.
2. deleteDisabledVariableIfUnused() checks if there are any workspace uses of the variable, after enable, this block is gone so the number of uses will be 0. If it is put before, the the disable() the number of uses will include this block and be 1.